gamemodes/terrortown/entities/entities/ttt_game_text.lua

```diff
@@ -52,30 +52,28 @@ end
 -- @return[default=true] boolean
 -- @realm shared
 function ENT:AcceptInput(name, activator)
-	if name == "Display" then
+	if name ~= "Display" then
+		return false
+	end
+				
+	if IsValid(activator) and activator:IsPlayer() then
 		local recv = activator
-
-		local r = self.Receiver
-		if r == RECEIVE_ALL then
-			recv = nil
-		elseif r == RECEIVE_DETECTIVE then
-			recv = GetRoleChatFilter(ROLE_DETECTIVE)
-		elseif r == RECEIVE_TRAITOR then
-			recv = GetTeamChatFilter(TEAM_TRAITOR)
-		elseif r == RECEIVE_INNOCENT then
-			recv = GetTeamChatFilter(TEAM_INNOCENT)
-		elseif r == RECEIVE_ACTIVATOR then
-			if not IsValid(activator) or not activator:IsPlayer() then
-				ErrorNoHalt("ttt_game_text tried to show message to invalid !activator\n")
-
-				return true
-			end
-		elseif r == RECEIVE_CUSTOMROLE and self.teamReceiver then
-			recv = GetTeamChatFilter(self.teamReceiver)
-		end
-
+		local receiver_tbl = {
+			RECEIVE_ALL = nil,
+			RECEIVE_DETECTIVE = GetRoleChatFilter(ROLE_DETECTIVE),
+			RECEIVE_TRAITOR = GetRoleChatFilter(TEAM_TRAITOR),
+			RECEIVE_INNOCENT = GetRoleChatFilter(TEAM_INNOCENT)
+		}
+			
+		recv = (self.teamReceiver) ? GetTeamChatFilter(self.teamReceiver) : receiver_tbl[self.Receiver]
```

Comment:
whoops, i just realized that the ternary operator doesn't exist in lua. (no clue why i thought it did ...)

But the following statement should be equivalent:
```suggestion
		recv = (self.teamReceiver) and GetTeamChatFilter(self.teamReceiver) or receiver_tbl[self.Receiver]
```